packages/kit/src/core/sync/create_manifest_data/index.js

```diff
@@ -286,12 +289,32 @@ export default function create_manifest_data({
 		? list_files({ config, dir: config.kit.files.assets, path: '' })
 		: [];
 
+	const params_base = path.relative(cwd, config.kit.files.params);
+
+	/** @type {Record<string, string>} */
+	const validators = {};
+	if (fs.existsSync(config.kit.files.params)) {
+		for (const file of fs.readdirSync(config.kit.files.params)) {
+			const ext = path.extname(file);
+			const type = file.slice(0, -ext.length);
+
+			if (/^[a-zA-Z_$][a-zA-Z0-9_$]*$/.test(type)) {
```

Comment:
if validator names can appear in filenames i'd forbid `$`

Reply:
working with them is just harder and they should be urlencoded too (which again could cause trouble with vite path to module resolution). sticking to characters that don't have special meaning in the contexts they are used avoids these issues.

Reply:
fair enough, have tightened it up. figured we really need to do it for parameter names as well on the same basis, which is technically a breaking change — added a changeset